### DISCUSSION TITLE
Using sherpa-onnx for languages different than Korean, Chinese and English

### DISCUSSION THREAD

**Discussion by ChrystianKacki** [k2-fsa/sherpa-onnx#1504] [Q&A]
On [this](https://k2-fsa.github.io/sherpa/onnx/pretrained_models) web site there are pre-trained models for languages like Chinese and English.
Is sherpa-onnx optimized only for languages from this web site or it is language independent, so I can create models for new languages like Polish ?

**Comment by Nasrudin78** [comment]
Any news on that topic? I'm also interested on Spanish language. 
I'm interested in KWS
Thx in advance

**Reply by csukuangfj** [reply]
You can create models for any language as long as you have enough training data and enough GPUs to train it.

Spanish is already supported. please read the doc carefully.

For your convenience this time, please see
https://k2-fsa.github.io/sherpa/onnx/pretrained_models/offline-transducer/nemo-transducer-models.html#sherpa-onnx-nemo-parakeet-tdt-0-6b-v3-int8-25-european-languages

There are many models in the doc supporting Spanish.

**Reply by Nasrudin78** [reply]
but for keyword spotting in a flutter app ? my intention is to trigger some action with a word spoken in Spanish. I've tested with english and works well for kws english model . But for spanish ? Could you provide some basic steps to do it properly?
Thanks a lot

**Reply by csukuangfj** [reply]
You need to train it by yourself for kws in this case. Please follow our icefall doc

**Reply by Nasrudin78** [reply]
Which of these recipes should I follow to train my model : https://k2-fsa.github.io/icefall/recipes/index.html ?
The one that is similar to my specs is ASR, but I need for KWS (since is a wakeupword for a Android app)...

**Reply by csukuangfj** [reply]
ASR and KWS are very similar. You can first read the ASR recipe and then adapt it to KWS.
